cmd/zc_traverser_blobfs.go

```diff
@@ -0,0 +1,124 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-azcopy/azbfs"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type blobFSTraverser struct {
+	rawURL    *url.URL
+	p         pipeline.Pipeline
+	ctx       context.Context
+	recursive bool
+
+	// Generic function to indicate that a new stored object has been enumerated
+	incrementEnumerationCounter func()
+}
+
+func newBlobFSTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context, recursive bool, incrementEnumerationCounter func()) (t *blobFSTraverser) {
+	t = &blobFSTraverser{
+		rawURL:                      rawURL,
+		p:                           p,
+		ctx:                         ctx,
+		recursive:                   recursive,
+		incrementEnumerationCounter: incrementEnumerationCounter,
+	}
+	return
+}
+
+func (t *blobFSTraverser) getPropertiesIfSingleFile() (*azbfs.PathGetPropertiesResponse, bool) {
+	pathURL := azbfs.NewFileURL(*t.rawURL, t.p)
+	pgr, err := pathURL.GetProperties(t.ctx)
+
+	if err != nil {
+		return nil, false
+	}
+
+	if pgr.XMsResourceType() == "directory" {
+		return pgr, false
+	}
+
+	return pgr, true
+}
+
+func (_ *blobFSTraverser) parseLMT(t string) time.Time {
```

Comment:
this should be in the SDK itself, but let's tolerate it since we'll remove it after interop